lib/ansible/modules/cloud/amazon/aws_secret.py

```diff
@@ -0,0 +1,358 @@
+        if desired_secret.kms_key_id != current_secret.get("KmsKeyId"):
+            return False
+        current_secret_value = self.client.get_secret_value(SecretId=current_secret.get("Name"))
+        if desired_secret.secret != current_secret_value.get(desired_secret.secret_type).encode("utf-8"):
+            return False
+        return True
+
+
+def rotation_match(desired_secret, current_secret):
+    """Compare secrets rotation configuration
+
+    Args:
+        desired_secret: camel dict representation of the desired secret state.
+        current_secret: secret reference as returned by the secretsmanager api.
+
+    Returns: bool
+    """
+    if desired_secret.rotation_enabled != current_secret.get("RotationEnabled"):
+        return False
+    if desired_secret.rotation_enabled:
+        if desired_secret.rotation_lambda_arn != current_secret.get("RotationLambdaARN"):
+            return False
+        if desired_secret.rotation_rules != current_secret.get("RotationRules"):
+            return False
+    return True
+
+
+def main():
+    module = AnsibleAWSModule(
+        argument_spec={
+            'name': dict(required=True),
+            'state': dict(choices=['present', 'absent'], default='present'),
+            'description': dict(),
+            'kms_key_id': dict(),
+            'secret_type': dict(choices=['binary', 'string'], default="string"),
+            'secret': dict(default=""),
+            'tags': dict(type='dict', default={}),
+            'rotation_lambda': dict(),
+            'rotation_interval': dict(type='int', default=30),
+            'recovery_window': dict(type='int', default=30),
+        },
+        supports_check_mode=True,
+    )
+
+    changed = False
+    state = module.params.get('state')
+    secrets_mgr = SecretsManagerInterface(module)
+    recovery_window = module.params.get('recovery_window')
+    secret = Secret(
+        module.params.get('name'),
+        module.params.get('secret_type'),
+        module.params.get('secret'),
+        description=module.params.get('description'),
+        kms_key_id=module.params.get('kms_key_id'),
+        tags=module.params.get('tags'),
+        lambda_arn=module.params.get('rotation_lambda'),
+        rotation_interval=module.params.get('rotation_interval')
+    )
+
+    current_secret = secrets_mgr.get_secret(secret.name)
+
+    if state == 'absent':
+        if current_secret:
+            if not current_secret.get("DeletedDate"):
+                result = secrets_mgr.delete_secret(secret.name, recovery_window=recovery_window)
+                changed = True
+            elif current_secret.get("DeletedDate") and recovery_window == 0:
+                result = secrets_mgr.delete_secret(secret.name, recovery_window=recovery_window)
+                changed = True
+        else:
+            result = "secret does not exist"
+
+    if state == 'present':
+        if current_secret is None:
+            result = secrets_mgr.create_secret(secret)
```

Comment:
Use camel_dict_to_snake_dict on result for consistency with other AWS modules. It looks like this is already happening for state == 'absent'

Reply:
indeed, done.